draft-js-dnd-plugin/src/index.js

```diff
@@ -0,0 +1,55 @@
+import cleanupEmpty from './modifiers/cleanupEmpty';
+import onDropFile from './modifiers/onDropFile';
+import onDropBlock from './modifiers/onDropBlock';
+import blockRendererFn from './blockRendererFn';
+import Image from './Image';
+import imageStyles from './imageStyles.css';
+import { Map } from 'immutable';
+import { EditorState } from 'draft-js';
+import EventEmitter from './utils/eventEmitter';
+import decorateComponentWithProps from 'decorate-component-with-props';
+
+const defaultTheme = Map({
+  imageWrapper: imageStyles.imageWrapper,
+  image: imageStyles.image,
+  imageButton: imageStyles.imageButton,
+
+  // This is for block-alignment-wrapper, only temporarily living here
+  left: imageStyles.left,
+  center: imageStyles.center,
+  right: imageStyles.right,
+});
+
+const uploadPlugin = (config = {}) => {
+  const theme = config.theme ? config.theme : defaultTheme;
+
+  const attachButtons = config.attachButtons !== false;
+
+  const emitter = new EventEmitter();
+  const blockRendererConfig = {
+    ...config,
+    emitter,
+    Image: decorateComponentWithProps(Image, { theme, attachButtons }),
+  };
+
+  return {
+    pluginProps: {
+      onChange: cleanupEmpty,
+      blockRendererFn: blockRendererFn(blockRendererConfig),
+      handleDroppedFiles: onDropFile(blockRendererConfig),
+      handleDrop: onDropBlock(blockRendererConfig),
+
+      // This is for block-alignment-wrapper, only temporarily living here
+      injectBlockProps: (block, getEditorState, onChange) => {
+        const editorState = getEditorState();
+        return {
+          refreshEditorState: () => onChange(EditorState.forceSelection(editorState, editorState.getCurrentContent().getSelectionAfter())),
+        };
+      },
+    },
+    addListener: emitter.addListener,
```

Comment:
Thats part of the event emitter to propagate plugin events (like upload progress).

Checkout these parts:
https://github.com/draft-js-plugins/draft-js-plugins/blob/dnd/docs/client/components/pages/Dnd/SimpleDndEditor/index.js#L60
https://github.com/draft-js-plugins/draft-js-plugins/blob/dnd/draft-js-dnd-plugin/src/components/dnd-wrapper.js#L14

I'm creating the plugin, and hand it to a decorator function as an argument, so it can hook into the upload progress event.

Of course, we could think about other ways, but only an event emitter would allow multiple targets to easily bind to the same event. Also it will allow to hook/unhook into events at any time. It will be particularly useful as soon as plugins start to depend on each other.

I'm really interested to hear what you @nikgraf and @juliankrispel think about that.